### PR TITLE
Combine CodeLanguageResolver with CodeLanguage

### PR DESCRIPTION
**Pull request by timcraft**
Merges CodeLanguageResolver methods into the CodeLanguage model (because it doesn't really add anything besides indirection). Removes a couple of related DocumentationConstraint methods.

### CODE REVIEW

File: spec/models/code_language_spec.rb [hunk 1]
```diff
@@ -43,4 +43,89 @@
...
+
+  describe '.find' do
+    it 'returns the code language for the given key' do
+      expect(CodeLanguage.find('ruby')).to be_kind_of(CodeLanguage)
+      expect(CodeLanguage.find('ios')).to be_kind_of(CodeLanguage)
+      expect(CodeLanguage.find('curl')).to be_kind_of(CodeLanguage)
+    end
+
+    context 'when given an invalid key' do
+      it 'returns nil' do
```
Comment (mheap) [inline spec/models/code_language_spec.rb]: The description doesn't seem to match the behaviour here

File: spec/models/code_language_spec.rb [hunk 2]
```diff
@@ -43,4 +43,89 @@
...
+
+  describe '.linkable' do
+    let(:languages) { CodeLanguage.linkable }
+
+    it 'returns an array of code languages' do
+      expect(languages).to be_kind_of(Array)
+      expect(languages[0]).to be_kind_of(CodeLanguage)
+    end
+
+    it 'excludes code languages that are not linkable' do
```
Comment (mheap) [inline spec/models/code_language_spec.rb]: It'd be good to assert that something that we know isn't linkable isn't in `languages` here

File: spec/models/code_language_spec.rb (hunk 1 above)
Comment (timcraft) [reply spec/models/code_language_spec.rb]: Yep, good catch 🙈

File: spec/models/code_language_spec.rb (hunk 2 above)
Comment (timcraft) [reply spec/models/code_language_spec.rb]: Added a couple of assertions.

**Pull request merged**